transport/graphsync/graphsync.go

```diff
@@ -149,19 +171,46 @@ func (t *Transport) OpenChannel(ctx context.Context,
 		exts = append(exts, doNotSendExt)
 	}
 
-	// Start tracking the data-transfer channel
 	ch := t.trackDTChannel(channelID)
 
+	// if we initiated this data transfer or the channel is open, run it immediately, otherwise
+	// put it in the request queue to rate limit DDOS attacks for push transfers
+	if channelID.Initiator == t.peerID || ch.isOpen {
+		// Start tracking the data-transfer channel
+
+		// Open a graphsync request to the remote peer
+		req, err := ch.open(ctx, channelID, dataSender, root, stor, doNotSendCids, exts)
```

Comment:
Suggest passing the `ch` object, rather than passing `channelID` here and then calling `getDTChannel(channelID)` in `runDeferredRequest`